### PR TITLE
Add Safe txs spell for polygon

### PR DESCRIPTION
**Pull request by tschubotz**
Brief comments on the purpose of your changes:
- Adds a spell for Safe transactions for polygon
- This PR is split out of #2754 

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by jeff-dude** [comment]
hi @tschubotz -- figured this was best place to contact you. the `safe_polygon_safes` spell just failed in prod for duplicates. for now, i've excluded the spell from future runs. can you check to see the issue and submit a fix? you should be able to query both the spell and compiled source query, find where the dupes are based on unique key columns

**Comment by tschubotz** [comment]
@jeff-dude I just looked into this.  

**tldr;** 
I believe Dune indexed the following 2 Polygon transactions wrong which lead to wrong traces in the traces table and hence to duplicated Safes in the Safes spell:

- `0xf44e71c59c4c5ec843da4c7867b5413cc9ddb7f7ce96ac971d7aeba46ed87c4b`
- `0x0b986ab5f2225f8384dfc68c5d769c1248466902592ea8d59f503c938eab171d`



Here's my line of thought, please let me know if I made a mistake:


### Check which addresses are duplicates
- I used: https://dune.com/queries/2037759
- Result:
  - `0xef43e05190d655515d46d9de1cc4db88a2e0f599` (I call it `D1`)
  - `0x9be479a5d988727f9548d1b134fd225398f0bef2` (I call it `D2`)

### Compare Dune traces for both addresses when retrieving Safe addresses 
== how the Safes spell is created, just filtered by D1 and D2, respectively.
- D1: https://dune.com/queries/2037164
  - Result: There are 2 100% identical traces in there, except for tx hash. tx index block hash.
  - the 2 tx hashes are:
      - `0x3280d1456920d668a4739bfe206a91d516fbee2fc6832ddd83c9c2d2bd8608c6` 👈  [This seems to be the actual Safe creation, also visible on Polygonscan.](https://polygonscan.com/tx/0x3280d1456920d668a4739bfe206a91d516fbee2fc6832ddd83c9c2d2bd8608c6)
      - `0xf44e71c59c4c5ec843da4c7867b5413cc9ddb7f7ce96ac971d7aeba46ed87c4b` 👈  [This one is suspicious and should not appear.](https://polygonscan.com/tx/0xf44e71c59c4c5ec843da4c7867b5413cc9ddb7f7ce96ac971d7aeba46ed87c4b)
- D2: https://dune.com/queries/2037173
  - Result: Same. There are 2 100% identical traces in there, except for tx hash. tx index block hash.
  - the 2 tx hashes are:
    - `0xabedbad7208ec72ea1c592d36c4dfadbc659917fd61b88a09e915ba6e2ef20eb` 👈  [This seems to be the actual Safe creation, also visible on Polygonscan. ](https://polygonscan.com/tx/0xabedbad7208ec72ea1c592d36c4dfadbc659917fd61b88a09e915ba6e2ef20eb)
    - `0x0b986ab5f2225f8384dfc68c5d769c1248466902592ea8d59f503c938eab171d`  👈  [This one is suspicious and should not appear.](https://polygonscan.com/tx/0x0b986ab5f2225f8384dfc68c5d769c1248466902592ea8d59f503c938eab171d)

This would mean that 2 setup calls to a single Safe were successful. That should not be possible. But who knows, there might be contract bugs, so let's check the suspicious transactions and their traces further .

- T1 (That's the suspicous tx from duplicate D1):  `0xf44e71c59c4c5ec843da4c7867b5413cc9ddb7f7ce96ac971d7aeba46ed87c4b`
- T2 (That's the suspicous tx from duplicate D2): `0x0b986ab5f2225f8384dfc68c5d769c1248466902592ea8d59f503c938eab171d`

### Check traces on Dune

I was unable to get a `select * from traces where tx_hash = x` running, so I just selected a handful columns.

- T1: https://dune.com/queries/2037541 
- T2: https://dune.com/queries/2037558

For both transactions, the first trace entry doesn't match with the `from` and `to` of the transaction. (Both according to Dune's tx table as well as e.g. Polygonscan.) That's suspicious.

### Compare Dune traces to traces fetched with other tool.

When checking the traces of the respective txs with an alternative tool, I get a different result which looks accurate to me. No Safe creation is part of the trace though.

- T2: https://openchain.xyz/trace/polygon/0xf44e71c59c4c5ec843da4c7867b5413cc9ddb7f7ce96ac971d7aeba46ed87c4b
- T2: https://openchain.xyz/trace/polygon/0x0b986ab5f2225f8384dfc68c5d769c1248466902592ea8d59f503c938eab171d


Hence my conclusion that Dune has at least T1 and T2 incorrectly indexed into the traces table. 

Could that be?

**Comment by tschubotz** [comment]
Seems Like Polygon Had a 157 block deep reorg yesterday, could that be related?

**Comment by jeff-dude** [comment]
thank you for the detailed analysis, much appreciated! let me run this by some of our internal data ingestion teammates.

**Comment by aalan3** [comment]
hey @tschubotz thank you for the detailed and well written analysis! this looks like a potential re-org issue. I'll look into this.

**Comment by tschubotz** [comment]
> hey @tschubotz thank you for the detailed and well written analysis! this looks like a potential re-org issue. I'll look into this.

Hey, @aalan3 have you been able to look into the re-org issue, yet? 
I would love to be able to get this PR merged :)

**Comment by aalan3** [comment]
@tschubotz sorry about the delay. This will get fixed next week, I'll ping you when it's done.

**Comment by bearishbuilder** [comment]
Hi @aalan3 is there any update on this?

**Comment by aalan3** [comment]
This should be fixed now! Let me know if you see any other issues.

**Comment by bearishbuilder** [comment]
Thanks @aalan3! Has it been merged and deployed to Dune tables?

**Comment by tschubotz** [comment]
@aalan3 thanks, can confirm that there are no duplicates anymore iin polygon.traces.

@jeff-dude safe_polygon.safes still contains the duplicates though (https://dune.com/queries/2037759 ) - how can we remove them?

**Comment by jeff-dude** [comment]
> @aalan3 thanks, can confirm that there are no duplicates anymore iin polygon.traces.
> 
> @jeff-dude safe_polygon.safes still contains the duplicates though (https://dune.com/queries/2037759 ) - how can we remove them?

i can remove the `safe_polygon.safes` spell from our exclusion list in prod, then it'll get back up-to-date. we can force a change to the spell in this PR in order for prod to automatically pick it up and refresh. i'll push a test change here

**Review by jeff-dude — APPROVED** [review]
should be set 👍
